imagetool/src/main/java/com/oracle/weblogic/imagetool/util/AruHttpHelper.java

```diff
@@ -0,0 +1,288 @@
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPathExpressionException;
+
+import com.oracle.weblogic.imagetool.installer.FmwInstallerType;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+public class AruHttpHelper {
+    private boolean success;
+    private Document results;
+    private String errorMessage;
+    private FmwInstallerType category;
+    private String version;
+    private String userId;
+    private String password;
+    private String release;
+
+    /**
+     * Constructor encapsulating the information for the ARU search.
+     *
+     * @param category of the product
+     * @param version of the product
+     * @param userId ARU user credentials
+     * @param password ARU password credentials
+     */
+    AruHttpHelper(FmwInstallerType category, String version, String userId, String password) {
+        this.category = category;
+        this.version = version;
+        this.userId = userId;
+        this.password = password;
+        this.success = true;
+        this.results = null;
+    }
+
+    AruHttpHelper(String userId, String password) {
+        this(null, null, userId, password);
+    }
+    
+    /**
+     * Get the error errorMessage.
+     *
+     * @return the search error message
+     */
+    String errorMessage() {
+        return errorMessage;
+    }
+
+    /**
+     * Set the error errorMessage.
+     *
+     * @param errorMessage message value.
+     */
+    AruHttpHelper setErrorMessage(String errorMessage) {
+        this.errorMessage = errorMessage;
+        return this;
+    }
+
+    /**
+     * Returns true if no conflicts ; false if there is conflicts.
+     * @return true if no conflicts ; false if there is conflicts
+     */
+    boolean success() {
+        return success;
+    }
+
+    /**
+     * Get the result object from the search.
+     *
+     * @return dom document detailing about the conflicts
+     */
+    Document results() {
+        return results;
+    }
+
+    /**
+     * Set the result object from the search.
+     *
+     * @param results dom document detailing about the conflicts
+     */
+    AruHttpHelper storeResults(Document results) {
+        this.results = results;
+        return this;
+    }
+
+    /**
+     * Return the FMWInstallerType category for the search.
+     *
+     * @return category
+     */
+    FmwInstallerType category() {
+        return category;
+    }
+
+    /**
+     * Return the product version number for the search.
+     *
+     * @return product version number
+     */
+    String version() {
+        return version;
+    }
+
+    /**
+     * Set the product version number for the search.
+     *
+     * @param version product version number
+     */
+    AruHttpHelper setVersion(String version) {
```

Comment:
I don't see this method used anywhere.  Delete?

Reply:
done